src/common/state/ColorControlPointList.C

```diff
@@ -1905,18 +1958,88 @@ ColorControlPointList::GetTagsAsString()
 // Creation:   Wed Jun  8 11:46:21 PDT 2022
 //
 // Modifications:
+//    Justin Privitera, Thu Aug 25 15:04:55 PDT 2022
+//    Made it a const function that takes a const arg.
 //
 // ****************************************************************************
 
 bool
-ColorControlPointList::HasTag(std::string tag)
+ColorControlPointList::HasTag(const std::string tag) const
 {
     for (int i = 0; i < tagNames.size(); i ++)
         if (tagNames[i] == tag)
             return true;
     return false;
 }
 
+// ****************************************************************************
+// Method: ColorControlPointList::GetTagIndex
+//
+// Purpose:
+//   Gets the index in the tag names vector of a given tag.
+//
+// Programmer: Justin Privitera
+// Creation:   Wed Aug 10 15:35:58 PDT 2022
+//
+// Modifications:
+//
+// ****************************************************************************
+
+int
+ColorControlPointList::GetTagIndex(const std::string tag) const
+{
+    for (int i = 0; i < tagNames.size(); i ++)
+        if (tagNames[i] == tag)
+            return i;
+    return -1;
+}
+
+// ****************************************************************************
+// Method: ColorControlPointList::ValidateTag
+//
+// Purpose: Make sure a tag conforms to rules. Returns a pair, where the first 
+//      element is a flag to say if the tag is good or not, and the second
+//      element is a string that contains an error message for use as needed.
+//
+// Programmer: Justin Privitera
+// Creation:   Thu Aug 25 10:05:45 PDT 2022
+//
+// Modifications:
+//
+// ****************************************************************************
+
+std::pair<bool, std::string>
+ColorControlPointList::ValidateTag(const std::string currtag) const
+{
+    auto success{false};
+    std::string outstr;
+    if (currtag == "")
+    {
+        outstr = "A tag for a color table "
+            "is an empty string. This tag will be discarded.\n";
+    }
+    // Check that the tag is alphanumeric or contains one of the allowed
+    // special characters.
+    else if (! std::all_of(currtag.begin(), currtag.end(), 
```

Comment:
I am having a lot trouble reading this block. It looks to me like the `else if` has two `{}` blocks...the first one indented and the second one. I have no idea what the first block is doing. I think its doing some kind of a lambda function over all characters (`std::all_of`) which winds up being true if all characters meet the conditions on lin `2026`. That is negated in the `else if` and the `outstr` is set to indicate its not valid. Ok, I think I understand it. now

Reply:
Is there a way I could reformat it to make it clearer?

Reply:
Hmmm...we don't really need to stick to 80 columns any more. I mean that is a hold over from the 1970's I think. Though we don't want to have it unlimited either. I think 132 wide is fine.

Maybe one of these....

```
    // Check that the tag is alphanumeric or contains one of the allowed
    // special characters.
    else if (! std::all_of(currtag.begin(), currtag.end(),[](char const &c) {return std::isalnum(c) || strchr(" -<>",c)!=0;}))
    {
        outstr = "The tag name \""
            + currtag + "\" is not valid. Tag names must contain "
            "only alphanumeric characters accompanied by the "
            "following 5 characters: \" \", \"-\", \"=\", \"<\", "
            "and \">\". This tag will be discarded.\n";
    }
```

or

```
    // Check that the tag is alphanumeric or contains one of the allowed
    // special characters.
    else if (! std::all_of(currtag.begin(), currtag.end(),
                   [](char const &c) {return std::isalnum(c) || strchr(" -<>",c)!=0;}))
    {
        outstr = "The tag name \""
            + currtag + "\" is not valid. Tag names must contain "
            "only alphanumeric characters accompanied by the "
            "following 5 characters: \" \", \"-\", \"=\", \"<\", "
            "and \">\". This tag will be discarded.\n";
    }
```

Reply:
ahh `strchr`. That's the magic function I was looking for. I'll do one of these.